umcontroller/umcontroller.go

```diff
@@ -400,6 +403,14 @@ func (umCtrl *Controller) UpdateComponents(
 	return umCtrl.currentComponents, umCtrl.updateError
 }
 
+func (umCtrl *Controller) SubscribeNewComponentListener() <-chan cloudprotocol.ComponentStatus {
+	listener := make(chan cloudprotocol.ComponentStatus)
+
+	umCtrl.newComponentListeners = append(umCtrl.newComponentListeners, listener)
```

Comment:
Slice appending is not thread safe operation: should be guarded with mutex.

Reply:
added mutex to Controller to guard newComponentListener